src/main/java/com/arpnetworking/clusteraggregator/client/HttpSourceActor.java

```diff
@@ -0,0 +1,330 @@
+                                    if (err instanceof InvalidRecordsException) {
+                                        BAD_REQUEST_LOGGER.debug()
+                                                .setMessage("Invalid records in http post")
+                                                .setThrowable(err)
+                                                .log();
+                                        sender.tell(HttpResponse.create().withStatus(400), self());
+                                    } else if (err instanceof NoRecordsException) {
+                                        BAD_REQUEST_LOGGER.debug()
+                                                .setMessage("No records in http post")
+                                                .setThrowable(err)
+                                                .log();
+                                        sender.tell(HttpResponse.create().withStatus(400), self());
+                                    } else {
+                                        BAD_REQUEST_LOGGER.warn()
+                                                .setMessage("Error handling http post")
+                                                .setThrowable(err)
+                                                .log();
+                                        sender.tell(HttpResponse.create().withStatus(500), self());
+                                    }
+                                }
+                            });
+                })
+                .build();
+    }
+
+    private static ImmutableMultimap<String, String> createHeaderMultimap(final Iterable<HttpHeader> headers) {
+        final ImmutableMultimap.Builder<String, String> headersBuilder = ImmutableMultimap.builder();
+
+        for (final HttpHeader httpHeader : headers) {
+            headersBuilder.put(httpHeader.lowercaseName(), httpHeader.value());
+        }
+
+        return headersBuilder.build();
+    }
+
+    private static com.arpnetworking.clusteraggregator.models.HttpRequest mapModel(
+            final Pair<ByteString, ImmutableMultimap<String, String>> pair) {
+        return new com.arpnetworking.clusteraggregator.models.HttpRequest(pair.second(), pair.first());
+    }
+
+    private List<AggregationMessage> parseRecords(final com.arpnetworking.clusteraggregator.models.HttpRequest request) throws IOException {
+        final ArrayList<AggregationMessage> records = Lists.newArrayList();
+        ByteString current = request.getBody();
+        Optional<AggregationMessage> messageOptional = AggregationMessage.deserialize(current);
+        while (messageOptional.isPresent()) {
+            final AggregationMessage message = messageOptional.get();
+            records.add(message);
+            current = current.drop(message.getLength());
+            messageOptional = AggregationMessage.deserialize(current);
+            if (!messageOptional.isPresent() && current.lengthCompare(4) > 0) {
+                LOGGER.trace()
```

Comment:
This log message is unnecessary.

Reply:
If you really want to capture the remaining bytes count, then please add it to the exception instance.

Reply:
I suppose.... if you like parsing strings out of exception messages....